From patchwork Tue Apr 18 14:21:09 2023
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sebastian Reichel <sebastian.reichel@collabora.com>
X-Patchwork-Id: 13215767
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 73FAFC6FD18
	for <linux-rockchip@archiver.kernel.org>;
 Tue, 18 Apr 2023 14:21:34 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=r3RyXVAVyAnZoTUlSUamrHtjQWlW+985Ld/q3wcWemU=; b=rbu+I75XmV+xyw
	X7hZyM3EeTs+9RPRlGsq6LOSmq8ttgapQgBTUcJwCKIbuTvFJBrBc/aOu87UZsFhFhxBfLayNkybr
	Ut01irE6NpnavMjuWD2gG52NlTF5IQgQjjnJv8l6mvBHpNTne1pF46JbStDXudYOtvaUYvJh9W1Cj
	TJiqEnciM5rbHj0/+yIVL+vQ82TCXAV0v3pVzaP50bFaVuTEcRhWlBQ75cKp6/0B+9s+pMDT6ao4M
	TV2Sxm2DNgagc1mr6/nWjiSpaz6eCfekKUwU/3WiQvdxxkE4YFCAA4c376pZ6Z0xL/0ekYqriHQFo
	tDrzAGILSLfW+OEEqpdw==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.96 #2 (Red Hat Linux))
	id 1pomCs-002PkD-0A;
	Tue, 18 Apr 2023 14:21:18 +0000
Received: from madras.collabora.co.uk ([46.235.227.172])
	by bombadil.infradead.org with esmtps (Exim 4.96 #2 (Red Hat Linux))
	id 1pomCp-002PiC-1p
	for linux-rockchip@lists.infradead.org;
	Tue, 18 Apr 2023 14:21:17 +0000
Received: from jupiter.universe (unknown [91.248.213.255])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange X25519 server-signature RSA-PSS (4096 bits) server-digest
 SHA256)
	(No client certificate requested)
	(Authenticated sender: sre)
	by madras.collabora.co.uk (Postfix) with ESMTPSA id 5B3AA6603218;
	Tue, 18 Apr 2023 15:21:11 +0100 (BST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=collabora.com;
	s=mail; t=1681827671;
	bh=JaiGX/zKLMMzlVpGom/T53NzObIEnY5d4OtYAaV22XU=;
	h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
	b=gwzQC1dQYfCTwHwTbf2svOb6fTO6TtWslcKeDLXMeV6PAyVuUGkLtnIQ9shBHdd0p
	 POD9tjKsdwX08hukajs9JDSmBLBPQl7ExeeUe0CpPN/uuqRMQck5z77DsdZhOuyGSG
	 bC/M/NPmt7tUSfbXiE/pMOJ6sPQUPtary/q5542c3KD/LPSJi+eFZ3q6psolhOU2/d
	 Mzo2Gtqq7Fe4Nhq4tp+0C29R0b3+CT8RSDlM/ltVS9n/K8wbhgvABaTP8u5iKVmXkF
	 dtSTqFks4b8boHqSORJ9qd1p8Cyyi0yVXCfBmNkXvhAnYQ0DsB2MMhFwBPTpkN0lov
	 fmoTUfFqe0OOg==
Received: by jupiter.universe (Postfix, from userid 1000)
	id 888D0480075; Tue, 18 Apr 2023 16:21:09 +0200 (CEST)
From: Sebastian Reichel <sebastian.reichel@collabora.com>
To: Marc Zyngier <maz@kernel.org>,
	Heiko Stuebner <heiko@sntech.de>
Cc: Rob Herring <robh+dt@kernel.org>,
	Krzysztof Kozlowski <krzysztof.kozlowski+dt@linaro.org>,
	Thomas Gleixner <tglx@linutronix.de>,
	Peng Fan <peng.fan@nxp.com>,
	Robin Murphy <robin.murphy@arm.com>,
	Peter Geis <pgwipeout@gmail.com>,
	XiaoDong Huang <derrick.huang@rock-chips.com>,
	Kever Yang <kever.yang@rock-chips.com>,
	linux-rockchip@lists.infradead.org,
	devicetree@vger.kernel.org,
	linux-kernel@vger.kernel.org,
	Sebastian Reichel <sebastian.reichel@collabora.com>,
	kernel@collabora.com
Subject: [PATCH v4 2/2] arm64: dts: rockchip: rk3588: add GIC ITS support
Date: Tue, 18 Apr 2023 16:21:09 +0200
Message-Id: <20230418142109.49762-3-sebastian.reichel@collabora.com>
X-Mailer: git-send-email 2.39.2
In-Reply-To: <20230418142109.49762-1-sebastian.reichel@collabora.com>
References: <20230418142109.49762-1-sebastian.reichel@collabora.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20230418_072115_844672_D2E284CE 
X-CRM114-Status: UNSURE (   9.19  )
X-CRM114-Notice: Please train this message.
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

Add the two Interrupt Translation Service (ITS) IPs that are part of the
GIC-600. They are mainly required for PCIe Message Signalled Interrupts
(MSI).

Co-developed-by: Kever Yang <kever.yang@rock-chips.com>
Signed-off-by: Kever Yang <kever.yang@rock-chips.com>
Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.com>
---
 arch/arm64/boot/dts/rockchip/rk3588s.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588s.dtsi b/arch/arm64/boot/dts/rockchip/rk3588s.dtsi
index 2124c654f665..62204b96b0b4 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588s.dtsi
@@ -1741,7 +1741,24 @@ gic: interrupt-controller@fe600000 {
 		mbi-alias = <0x0 0xfe610000>;
 		mbi-ranges = <424 56>;
 		msi-controller;
+		ranges;
+		#address-cells = <2>;
 		#interrupt-cells = <4>;
+		#size-cells = <2>;
+
+		its0: msi-controller@fe640000 {
+			compatible = "arm,gic-v3-its";
+			msi-controller;
+			#msi-cells = <1>;
+			reg = <0x0 0xfe640000 0x0 0x20000>;
+		};
+
+		its1: msi-controller@fe660000 {
+			compatible = "arm,gic-v3-its";
+			msi-controller;
+			#msi-cells = <1>;
+			reg = <0x0 0xfe660000 0x0 0x20000>;
+		};
 
 		ppi-partitions {
 			ppi_partition0: interrupt-partition-0 {
